Gems/RecastNavigation/Code/Source/Components/RecastNavigationMeshCommon.cpp

```diff
@@ -0,0 +1,349 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include "RecastNavigationMeshCommon.h"
+
+#include <DetourNavMeshBuilder.h>
+#include <AzCore/Console/Console.h>
+#include <AzCore/Debug/Profiler.h>
+#include <AzFramework/Entity/EntityDebugDisplayBus.h>
+#include <AzFramework/Physics/PhysicsScene.h>
+#include <RecastNavigation/RecastNavigationSurveyorBus.h>
+
+AZ_DEFINE_BUDGET(Navigation);
+
+namespace RecastNavigation
+{
+    NavigationTileData RecastNavigationMeshCommon::CreateNavigationTile(TileGeometry* geom,
```

Comment:
comment: This is a bit of a monster function 😛 the sections are super helpful but we could potentially break this up into some smaller functions if possible (maybe outside of the scope of this PR as an initial version)

Reply:
yeah, good idea, in the future this will be broken up